src/ui/opengl/castlecontrols_edit.inc

```diff
@@ -361,4 +362,16 @@ begin
     Result := inherited CapturesEventsAtPosition(Position);
 end;
 
+procedure TCastleEdit.SetFocused(const Value: boolean);
+begin
+  inherited;
+
+  {$ifdef ANDROID}
+  if Value then
+    Messaging.Send(['change-keyboard-state', 'on'])
+  else
```

Comment:
Use TMessaging.BoolToStr to send a boolean to a message. So this can be just

```
Messaging.Send((['change-keyboard-state', TMessaging.BoolToStr(Value)]);
```

There is an equivalent method in Java to decode the boolean:)